src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -124,31 +129,70 @@ protected void Dispose(bool disposing)
         /// Makes a TestCase from an NUnit test, adding
         /// navigation data if it can be found.
         /// </summary>
-        private TestCase MakeTestCaseFromNUnitTest(ITest nunitTest)
+        private TestCase MakeTestCaseFromXmlNode(XmlNode testNode)
         {
-            //var testCase = MakeTestCaseFromTestName(nunitTest.TestName);
             var testCase = new TestCase(
-                                     nunitTest.TestName.FullName,
+                                     testNode.GetAttribute("fullname"),
                                      new Uri(NUnitTestExecutor.ExecutorUri),
                                      this.sourceAssembly)
             {
-                DisplayName = nunitTest.TestName.Name,
+                DisplayName = testNode.GetAttribute("name"),
                 CodeFilePath = null,
                 LineNumber = 0
             };
 
-            var navData = GetNavigationData(nunitTest.ClassName, nunitTest.MethodName);
+            var navData = GetNavigationData(GetClassName(testNode), GetMethodName(testNode));
             if (navData != null)
             {
                 testCase.CodeFilePath = navData.FileName;
                 testCase.LineNumber = navData.MinLineNumber;
             }
 
-            testCase.AddTraitsFromNUnitTest(nunitTest);
+            testCase.AddTraitsFromTestNode(testNode);
 
             return testCase;
         }
 
+        private static string GetClassName(XmlNode node)
+        {
+#if true
+            return node.GetAttribute("classname");
+#else
```

Comment:
What is the point of all this unused code? Just copied in? Also, should these two methods be extension methods on XmlNode?